google-datacatalog-sisense-connector/src/google/datacatalog_connectors/sisense/addons/elasticube_dependency_finder.py

```diff
@@ -0,0 +1,390 @@
+
+        Returns:
+            dict: Keys are the names and values their corresponding entries.
+        """
+        entries_dict = {}
+        for entry_name in entry_names or []:
+            entries_dict[entry_name] = self.__datacatalog_facade.get_entry(
+                entry_name)
+
+        return entries_dict
+
+    def __list_tags(self, entry_names: List[str]) -> Dict[str, List[Tag]]:
+        """Get Data Catalog Tags for a given list of entry names.
+
+        Returns:
+            dict: Keys are the entry names and values are lists that contains
+            all their related tags.
+        """
+        tags_dict = {}
+        for entry_name in entry_names or []:
+            tags_dict[entry_name] = self.__datacatalog_facade.list_tags(
+                entry_name)
+
+        return tags_dict
+
+    @classmethod
+    def __filter_relevant_tags(cls, entry: Entry, tags: List[Tag], table: str,
+                               column: str) -> List[Tag]:
+        """Filter Tags for ElastiCube dependencies finding. "Relevant", in this
+        context, means Tags that have ElastiCube-related information such as
+        data source, table, or column names.
+
+        Returns:
+            list: The filtered Tags.
+        """
+        filtered_tags = []
+
+        asset_metadata_tag = cls.filter_asset_metadata_tag(tags)
+        if asset_metadata_tag:
+            filtered_tags.append(asset_metadata_tag)
+        table_column_matching_tags = cls.__filter_table_column_matching_tags(
+            table, column, tags)
+        filtered_tags.extend(
+            cls.__sort_tags_by_schema(entry.schema.columns, '',
+                                      table_column_matching_tags))
+
+        return filtered_tags
+
+    @classmethod
+    def filter_asset_metadata_tag(cls, tags: List[Tag]) -> Optional[Tag]:
+        """Filter Dashboard and Widget metadata enrichment Tags created from
+        Templates identified by the `TAG_TEMPLATE_ID_DASHBOARD` and
+        `TAG_TEMPLATE_ID_WIDGET` constants.
+
+        Returns:
+            list: The filtered Tags.
+        """
+        template_ids = [
+            constants.TAG_TEMPLATE_ID_DASHBOARD,
+            constants.TAG_TEMPLATE_ID_WIDGET
+        ]
+
+        for tag in tags or []:
+            template_id = re.match(pattern=cls.__TAG_TEMPLATE_NAME_PATTERN,
+                                   string=tag.template).group('id')
+
+            if template_id in template_ids:
+                return tag
+
+    @classmethod
+    def __filter_table_column_matching_tags(cls, table: str, column: str,
+                                            tags: List[Tag]) -> List[Tag]:
+        """Filter JAQL-related Tags which `table` and/or `column` fields match
+        the provided values.
+
+        If both `table` and `column` args are provided, a given Tag's
+        corresponding fields must match both values in order to added to the
+        resulting list.
```

Comment:
small typo, in order to **be** added to the resulting list.

Reply:
Fixed, thanks for catching this.